src/searchbar.cpp

```diff
@@ -233,6 +242,7 @@ void SearchBarLineEdit::updateCompletion()
         return;
     }
     m_token++;
+    m_suggestionsAreValidFor = m_searchbarInput;
```

Comment:
Is this the correct place to set `m_suggestionsAreValidFor`? Shouldn't it rather be set in `onInitialSuggestions()`? Of course it will slightly complicate the solution since that information will have to be somehow passed to `onInitialSuggestions()`, however we cannot compromise correctness. A workaround may be to set it in the handler of the `SuggestionListWorker::searchFinished()` signal in `fetchSuggestions()`.

Reply:
Done